tests/conftest.py

```diff
@@ -187,11 +193,10 @@ def _is_generic_test(item) -> bool:
 
 
 def _generic_cloud(config) -> str:
-    c = config.getoption('--generic-cloud')
-    cloud_to_run = _get_cloud_to_run(config)
-    if c not in cloud_to_run:
-        c = cloud_to_run[0]
-    return c
+    generic_cloud_option = config.getoption('--generic-cloud')
+    if generic_cloud_option is not None:
+        return generic_cloud_option
```

Comment:
If an invalid cloud is passed to `--generic-cloud`, previously we had an implicit fallback to a valid cloud from `_get_cloud_to_run`. Looks like that's removed now. Should we retain that behavior?

Reply:
I think we should throw in this case. Will update.

Reply:
Ah - actually we are protected against this because the CLI parser will only allow valid clouds.
https://github.com/skypilot-org/skypilot/blob/914328acb8269d79e304ad891f84d220e077565c/tests/conftest.py#L73-L77